### DISCUSSION TITLE
Using Smart Connections with Text Generator (another Obsidian Plugin)

### DISCUSSION THREAD

**Discussion by neojininja** [brianpetro/obsidian-smart-connections#551] [Q&A] [answered]
Hello! I'm very interested in learning how to use Smart Connections with the Text Generator Plugin. Someone has already made an attempt at this but it's not working for me. I've provided error screenshots below and also the reference information I used. Any insight to make this work is greatly appreciated! Thank you in advance.

![image1](https://github.com/brianpetro/obsidian-smart-connections/assets/130234091/bfcc1cab-1974-463a-8b4e-8a98ad3d3873)

![image2](https://github.com/brianpetro/obsidian-smart-connections/assets/130234091/8a31933a-065b-472f-a09c-2769aae0e33d)

![image3](https://github.com/brianpetro/obsidian-smart-connections/assets/130234091/e4b29b1f-c5b6-4909-b4ba-0c360740030b)

**Here's a quote from someone who got this working on Discord**
---
To contribute with updated code request, we will take the original script provided and modify it according to the suggestion made. The modification involves updating a specific API call within the script to ensure compatibility. Here's how the revised script looks:

```javascript
const query = this.tg_selection;
const results = await app.plugins.plugins["smart-connections"].api.search(query);
const contextResults = await Promise.all(
  results.slice(0, 5).map(r => 
    app.plugins.plugins["smart-connections"].brain.smart_blocks.get(r.link, { lines: 10, max_chars: 1000 }).get_as_context_for_chat()
  )
);
const context = contextResults.join("\n---\n");

const summary = await gen(`
Query: 
${query}

Outcomes: 
${context}

Prompt:
Synthesize the results section, highlighting the relationship between the initial query and the outcomes. Structure the content using Markdown formatting to establish a clear information hierarchy. This should include titles, subtitles, and varying font sizes. For improved readability, **bolden all verbs** and specific action-oriented phrases. Ensure that the Markdown elements are properly used to organize the content systematically.

Summary: 

`)

return summary;
```
---

**And here's the one in Github:**

Source: https://github.com/text-gen/tg-smartConnections/blob/main/prompts/summarize.md
---
promptId: summarize
name: 🗞️Summarize The Results of Smart Connections
description: summarize
author: Noureddine
tags:
  - smart-connections
version: 0.0.1
disableProvider: true
provider:
---
{{#script}}
```js
if(!app.plugins.plugins["smart-connections"]) 
	return error("Smart connection plugin not installed")
```
{{/script}}
***
This input template is currently disabled due to the 'disabledProvider' setting being set to true.

If you wish to utilize this template with a provider, such as Chatbot or another service, please follow these steps:
- Enable the provider by setting 'disabledProvider' to false.
- Cut and paste everything from the output template into this section.
- Replace the content in the output template with '{{output}}'.
- Remember to delete this instruction text.
***
{{#script}}
```js
const query = this.tg_selection;
const results = await app.plugins.plugins["smart-connections"].api.search(query);
const contextResults = await Promise.all(
  results.slice(0, 5).map(r => 
    app.plugins.plugins["smart-connections"].file_retriever(r.link, { lines: 10, max_chars: 1000 })
  )
);
const context = contextResults.join("\n---\n");

const summary = await gen(`
Query: 
${query}

Outcomes: 
${context}

Prompt:
Synthesize the results section, highlighting the relationship between the initial query and the outcomes. Structure the content using Markdown formatting to establish a clear information hierarchy. This should include titles, subtitles, and varying font sizes. For improved readability, bolden all verbs and specific action-oriented phrases. Ensure that the Markdown elements are properly used to organize the content systematically.

Summary: 

`)

return summary;
```
{{/script}}

**Comment by stracker-phil** [comment]
For me, the latest version of Smart-Connections requires a slightly different template:

**Original:**
```js
...
const contextResults = await Promise.all(
  results.slice(0, 5).map(r => 
    app.plugins.plugins["smart-connections"].file_retriever(r.link, { lines: 10, max_chars: 1000 })
  )
);
...
```

**Working:**
```js
...
const contextResults = await Promise.all(
  results.slice(0, 5).map(r => 
    app.plugins.plugins["smart-connections"]
      .brain.smart_blocks.get(r.link, { lines: 10, max_chars: 1000 })
      .get_as_context()
  )
);
...
```

**Comment by Cody-W-Tucker** [comment]
Here's my current working version, haven't thought of a better way to filter out the current note's from the smart blocks, but here's a simple way.

```Javascript
const query = this.tg_selection;
const results = await window['SmartSearch'].search(query);

// Get the source_key of the current note (assuming it's the first result in the array)
const currentNoteSourceKey = results[0].source_key; // Change this if the current note isn't always the first result

// Filter out results that have the same source_key as the current note
const filteredResults = results.filter(r => r.source_key !== currentNoteSourceKey);

const contextResults = await Promise.all(
  filteredResults
  .slice(0, 3).map(r => 
    app.plugins.plugins["smart-connections"]
    .env.smart_blocks.get(r.key)
    .get_as_context()
  )
);

const context = contextResults.join('\n\n');
```

**Comment by brianpetro** [comment, marked as answer]
See this for integrating Smart Environment with Templater: https://github.com/brianpetro/obsidian-smart-connections/discussions/1081#discussioncomment-13348167
